### ISSUE TITLE
[BUG] The implementation of TensorFlow optimizers for distributed training is incorrect

### ISSUE THREAD

**Issue by edwardyehuang** [keras-team/keras#19245]
The implementation of TensorFlow optimizers for distributed training is incorrect.

https://github.com/keras-team/keras/blob/24d9e27e6d118b3ff23a4ba81cdea6d12e6c0551/keras/backend/tensorflow/optimizer.py#L102-L124

The `learning_rate` passed to line 115 is already `PerReplica`, instead of python object or scalar because of line 107.

Here is the correct implementation:

```python
 def _distributed_tf_update_step(
        self, distribution, grads_and_vars, learning_rate
    ):
        def apply_grad_to_update_var(var, grad, lr):
            return self.update_step(grad, var, lr)

        for grad, var in grads_and_vars:
            distribution.extended.update(
                var, apply_grad_to_update_var, args=(grad,learning_rate,), group=False
            )
```

**Comment by fchollet** [comment]
This should not affect any behavior since the learning rate does not depend on the replica. Also, we've updated that part of the code. Thanks for the report.
